puffin_http/README.md

```diff
@@ -8,3 +8,46 @@
 A HTTP server/client for communicating [`puffin`](https://github.com/EmbarkStudios/puffin) profiling events.
 
 You can view them using [`puffin_viewer`](https://github.com/EmbarkStudios/puffin/tree/main/puffin_viewer).
+
+# Architecture
+## server
+Listens for incoming connections and streams them puffin profiler data.
+- struct Server
```

Comment:
Lets keep architectural docs in [rust docs](https://doc.rust-lang.org/rust-by-example/meta/doc.html), no need to manually and separately document code